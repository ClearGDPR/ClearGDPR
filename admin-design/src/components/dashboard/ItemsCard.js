import React from 'react';
import Card from './Card';
import GraphBar from './GraphBar';

const ItemsCard = props => {
  return (
    <Card
      size={props.size}
      cols={props.cols}
      title={props.title}
      togglePanel={props.togglePanel}
    >
      <div className="row items-container">
        {props.data.map((processor, id) => {
          return (
            <div className="item" key={id}>
              <p className="number">{processor.consented}%</p>
              <GraphBar width={processor.consented} />
              <p className="name">{processor.name}</p>
            </div>
          );
        })}
      </div>
    </Card>
  );
};

export default ItemsCard;
