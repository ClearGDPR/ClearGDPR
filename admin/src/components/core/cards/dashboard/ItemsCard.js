import React from 'react';
import PropTypes from 'prop-types';

import Card from 'components/core/cards/dashboard/Card';
import GraphBar from 'components/core/cards/dashboard/GraphBar';

const ItemsCard = props => {
  return (
    <Card size={props.size} cols={props.cols} title={props.title} onClick={props.onClick}>
      <div className="row items-container">
        {props.data.map(({ name, fillPercent }, id) => {
          return (
            <div className="item" key={id}>
              <p className="number">{fillPercent}%</p>
              <GraphBar width={fillPercent} />
              <p className="name">{name}</p>
            </div>
          );
        })}
      </div>
    </Card>
  );
};

ItemsCard.propTypes = {
  size: PropTypes.number,
  cols: PropTypes.number,
  data: PropTypes.arrayOf(
    PropTypes.shape({
      fillPercent: PropTypes.number,
      name: PropTypes.string
    })
  ),
  onClick: PropTypes.func,
  title: PropTypes.string
};

export default ItemsCard;
