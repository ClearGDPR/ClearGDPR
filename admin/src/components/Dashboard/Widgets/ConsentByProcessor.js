import React from 'react';
import PropTypes from 'prop-types';
import ItemsCard from 'components/core/cards/dashboard/ItemsCard';

const ConsentByProcessor = ({ processors, totalSubjects }) => {
  return (
    <ItemsCard
      data={processors.map(p => ({
        name: p.name,
        fillPercent: totalSubjects ? Math.floor(p.consented / totalSubjects * 100) : 100
      }))}
    />
  );
};

ConsentByProcessor.propTypes = {
  processors: PropTypes.array,
  totalSubjects: PropTypes.number
};

export default ConsentByProcessor;
