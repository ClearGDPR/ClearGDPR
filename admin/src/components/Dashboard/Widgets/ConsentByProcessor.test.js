import React from 'react';
import { shallow } from 'enzyme';
import ConsentByProcessor from './ConsentByProcessor';
import toJson from 'enzyme-to-json';
import ItemsCard from 'components/core/cards/dashboard/ItemsCard';

describe('(Component) ConsentByProcessor', () => {
  it('should render', () => {
    const props = {
      totalSubjects: 100,
      processors: [
        {
          name: 'p1',
          consented: 1
        }
      ]
    };
    const component = shallow(<ConsentByProcessor {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render if numbers are 0', () => {
    const props = {
      totalSubjects: 100,
      processors: [
        {
          name: 'p1',
          consented: 1
        }
      ]
    };
    const component = shallow(<ConsentByProcessor {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should do the math correctly', () => {
    const props = {
      totalSubjects: 100,
      processors: [
        {
          name: 'p1',
          consented: 32
        }
      ]
    };
    const component = shallow(<ConsentByProcessor {...props} />);
    expect(component.find(ItemsCard).props().data[0].fillPercent).toEqual(32);
    expect(toJson(component)).toMatchSnapshot();
  });
});
