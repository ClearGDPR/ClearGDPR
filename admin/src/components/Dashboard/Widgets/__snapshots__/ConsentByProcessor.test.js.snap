// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ConsentByProcessor should do the math correctly 1`] = `
<ItemsCard
  cols={4}
  data={
    Array [
      Object {
        "fillPercent": 32,
        "name": "p1",
      },
    ]
  }
/>
`;

exports[`(Component) ConsentByProcessor should render 1`] = `
<ItemsCard
  cols={4}
  data={
    Array [
      Object {
        "fillPercent": 1,
        "name": "p1",
      },
    ]
  }
/>
`;

exports[`(Component) ConsentByProcessor should render if numbers are 0 1`] = `
<ItemsCard
  cols={4}
  data={
    Array [
      Object {
        "fillPercent": 100,
        "name": "p1",
      },
    ]
  }
/>
`;
